code/data/multi_country/convert-policies-raw-to-interim.py

```diff
@@ -0,0 +1,150 @@
+import pandas as pd
+import numpy as np
+import src.utils as cutil
+import json
+import operator
+import argparse
+
+
+def policy_conversion_parser():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--c",
+        choices=sorted(cutil.ISOS),
+        default=None,
+        type=str.upper,
+        help="Country ISO code",
+    )
+
+    parser.add_argument("--l", default=False, action="store_true", help="Print logs")
+
+    return parser
+
+
+parser = policy_conversion_parser()
+
+args = parser.parse_args()
+print_logs = args.l
+
+
+def get_country_list(country_input):
+    if country_input != None:
+        return [country_input]
+    return cutil.ISOS
+
+
+country_list = get_country_list(args.c)
+
+countries_wo_intensity = ["CHN", "IRN"]
+
+op_dict = {
+    ">": operator.gt,
+    "<": operator.lt,
+    ">=": operator.ge,
+    "<=": operator.le,
+    "=": operator.eq,
+}
+
+
+def apply_rule(df, src_policy, op_str, src_val, dst_rule, country_code):
+
+    dst_policy, dst_val = dst_rule
+
+    # Get operator function from operator string
+    op = op_dict[op_str]
+
+    if print_logs:
+        print(
+            f"{country_code}: where {src_policy} {op_str} {src_val}, set {dst_policy} = {dst_val}"
+        )
+
+    mask = df["policy"] == src_policy
+    mask = (mask) & (df["optional"] != "Y")
+    if country_code not in countries_wo_intensity:
+        mask = (mask) & (op(df["policy_intensity"], src_val))
+
+    pcopy = df[mask].copy()
+
+    pcopy["policy"] = dst_policy
+
+    if country_code not in countries_wo_intensity:
+        pcopy["policy_intensity"] = dst_val
+
+    df = pd.concat([df, pcopy], ignore_index=True).sort_values(
+        "date_start", ascending=True
+    )
+
+    return df
+
+
+def apply_usa_rule(df, src_policy, dst_policies):
```

Comment:
Are we handling trailing spaces well here? Just confirming. Also should we drop anything that has null intensity_group within this function? (note that when I called `pd.read_csv` on the usa policy csv, one row that was NA was actually converted to `np.nan` but others were not (becaues they had trailing spaces, e.t. "NA ")

Reply:
good points!

Reply:
I tried to handle these issues in `clean_intensities_usa(df)`